From 1ff012598d5827887597b7ff3660c60898f05eaf Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 21 Dec 2023 15:52:42 +0000
Subject: [PATCH 0803/1085] dts: bcm271-rpi-3-b: Make brcmvirt-gpio a firmware
 child

The driver makes use of the fact that the firmware node is its parent,
so we'd better make it so.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/bcm2710-rpi-3-b.dts | 18 +++++++-----------
 1 file changed, 7 insertions(+), 11 deletions(-)

--- a/arch/arm/boot/dts/broadcom/bcm2710-rpi-3-b.dts
+++ b/arch/arm/boot/dts/broadcom/bcm2710-rpi-3-b.dts
@@ -168,17 +168,6 @@
 	};
 };
 
-&soc {
-	virtgpio: virtgpio {
-		compatible = "brcm,bcm2835-virtgpio";
-		gpio-controller;
-		#gpio-cells = <2>;
-		firmware = <&firmware>;
-		status = "okay";
-	};
-
-};
-
 &firmware {
 	expgpio: expgpio {
 		compatible = "raspberrypi,firmware-gpio";
@@ -194,6 +183,13 @@
 				  "PWR_LOW_N";
 		status = "okay";
 	};
+
+	virtgpio: virtgpio {
+		compatible = "brcm,bcm2835-virtgpio";
+		gpio-controller;
+		#gpio-cells = <2>;
+		status = "okay";
+	};
 };
 
 &uart0 {
